numpy/_core/arrayprint.py

```diff
@@ -85,7 +85,7 @@ def _make_options_dict(precision=None, threshold=None, edgeitems=None,
     if sign not in [None, '-', '+', ' ']:
         raise ValueError("sign option must be one of ' ', '+', or '-'")
 
-    if legacy == False:
+    if legacy is False:
```

Comment:
This changes behaviour for `legacy` equal to `O`. Maybe it is ok in this case, but I am a bit surprised the change end up here

Reply:
Changing `x == False` → `x is False` works when the `x` is a `bool`. I am not certain automatic fixes check that. In any case, comparing a non-`bool` to `False` is poor style that needs to be fixed.

Reply:
I'll leave out such fixes for now.

Reply:
For [numpy.array](https://numpy.org/doc/stable/reference/generated/numpy.array.html), `== None` and `is None` are not equivalent. Therefore, the ruff/pycodestyle rule [E711](https://docs.astral.sh/ruff/rules/none-comparison/) cannot be applied.

Reply:
Did you mean to revert this change?